# Week 08

### Topic
 1. Cars data & using a t-Test
 1. Extra exercise with cars data

Built from: http://www.r-tutor.com/elementary-statistics/inference-about-two-populations/population-mean-between-two-independent-samples


In [1]:
# resize
require(repr)
options(repr.plot.width=4, repr.plot.height=4)

Loading required package: repr


What datasets are available that "come with" R?

In [2]:
data()

## 1. Cars data & using a t-Test

For this, we are going to use the `mtcars` dataset, so lets load this:

In [3]:
data(mtcars)

Check out header & part of dataset:

In [4]:
head(mtcars, 6)

mpg  cyl disp hp  drat wt    qsec  vs am gear carb
Mazda RX4         21.0 6   160  110 3.90 2.620 16.46 0  1  4    4   
Mazda RX4 Wag     21.0 6   160  110 3.90 2.875 17.02 0  1  4    4   
Datsun 710        22.8 4   108   93 3.85 2.320 18.61 1  1  4    1   
Hornet 4 Drive    21.4 6   258  110 3.08 3.215 19.44 1  0  3    1   
Hornet Sportabout 18.7 8   360  175 3.15 3.440 17.02 0  0  3    2   
Valiant           18.1 6   225  105 2.76 3.460 20.22 1  0  3    1

So, what do all of these mean?  We can figure this out with the `help` function:

In [5]:
help(mtcars)

So we see that the first colum is miles/gallon then we have number of cylinders, displacement (no idea), horsepower, real axel ratio (again, not sure), how fast its acceleration is (I assume) engine shape, wheter its manual or automatic transmission, number of gears and number of carbureators.

**Question:** Is there a difference in fuel efficiency (`mpg`) between automatic and manual (stick) transmission cars?

For this exercise, what we want to remember from the help:
  * parameter `am` $\rightarrow$ automatic or manual
  * 0=automatic, 1=manual

Grab mpg for manual & auto:

In [6]:
mpg_manual = subset(mtcars$mpg, mtcars$am == 1)
mpg_auto = subset(mtcars$mpg, mtcars$am == 0)

Let's look at these entries:

In [7]:
mpg_manual

[1] 21.0 21.0 22.8 32.4 30.4 33.9 27.3 26.0 30.4 15.8 19.7 15.0 21.4

In [8]:
mpg_auto

[1] 21.4 18.7 18.1 14.3 24.4 22.8 19.2 17.8 16.4 17.3 15.2 10.4 10.4 14.7 21.5
[16] 15.5 15.2 13.3 19.2

Note that they have different sizes -- we want to compare the means of these two different sized vectors.

Let's use a t-test to see if the means are different:

In [10]:
t.test(mpg_auto, mpg_manual)


	Welch Two Sample t-test

data:  mpg_auto and mpg_manual
t = -3.7671, df = 18.332, p-value = 0.001374
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -11.280194  -3.209684
sample estimates:
mean of x mean of y 
 17.14737  24.39231 


OK, what does this say?  Le'ts go through it:
  * we can say with a 95% confidence the difference between the means is from 3.2 - 11.3 mpg
  * note it says: "true difference in means is not equal to 0" so auto != manual
  * this was a 2 sided test
  
Let's say we had reason to believe that mpg_auto < mpg_manual.  Alternative: that "x" has a greater mean than "y":

In [11]:
print(t.test(mpg_auto, mpg_manual,alternative="greater"))


	Welch Two Sample t-test

data:  mpg_auto and mpg_manual
t = -3.7671, df = 18.332, p-value = 0.9993
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -10.57662       Inf
sample estimates:
mean of x mean of y 
 17.14737  24.39231 



Can check out p-value specifically:

In [12]:
print('p-value')
print(t.test(mpg_auto, mpg_manual,alternative="greater")$p.value)

[1] "p-value"
[1] 0.9993132


We can see this is >> 0.05. This makes sense since manual cars of this era generally get better mpg than automatics.  Note: here we assumed we had *some reason* to believe manual cars got worse gas milage.

## 2. Extra exercise with cars data

Extension:
 * Part 1: Select for engine shape - are the mean mpg's the same? 
 * Part 2: While the number of samples is too small, we might be tempted to do this calculation with a normal distribution.  Re-do this test using the normal distribution. Recall for the difference of two means: SE = $\left( \frac{SD_1^2}{n_1} + \frac{SD_2^2}{n_2} \right)^{0.5}$


#### Part 1

In [13]:
mpg_v = subset(mtcars$mpg, mtcars$vs == 0)
mpg_s = subset(mtcars$mpg, mtcars$vs == 1)

print(t.test(mpg_v, mpg_s))


	Welch Two Sample t-test

data:  mpg_v and mpg_s
t = -4.6671, df = 22.716, p-value = 0.0001098
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -11.462508  -4.418445
sample estimates:
mean of x mean of y 
 16.61667  24.55714 



p-value = 0.0001098 << 0.05 $\rightarrow$ so, they are different

#### Part 2

In [14]:
SE_combined = (sd(mpg_v)^2/length(mpg_v) + sd(mpg_s)^2/length(mpg_s))^0.5
Zscore = (mean(mpg_v)-mean(mpg_s))/SE_combined # -4.66
pvalue = 2.0 * (pnorm(Zscore)) # 2* for 2 sided test
print(pvalue)

[1] 3.055509e-06


Note: if did mean(mpg_s)-mean(mpg_v), then: * pnorm(Zscore,lower.tail=F)

 
#### BONUS:
Plot all data with "pairs" function any relationships you want to explore?

For example: means of mpg for different #'s of cylinders or, horse power as a function of different #'s of cylinders, or auto vs. manual. Or, what is the probability that hp of autos are higher than manuals? t-test & normal model. Or, make a cut for "low and high" horse power, and see how mpg depends on this.